css-overflow-5/Overview.bs

```diff
@@ -938,47 +936,51 @@ Issue: Explore whether scrolltarget can be more directly associated with anchor
-		1. For each <var>target</var> in <var>targets</var>:
-			1. Set <var>selected</var> to <var>target</var>.
-			1. Let <var>targetPosition</var> be the position that would be scrolled to if we scroll <var>target</var> into view.
-			1.	: If <var>targetPosition</var>'s scroll block offset is less than or equal to <var>position</var>'s scroll block offset, and
-					<var>targetPosition</var>'s scroll inline offset is less than or equal to <var>position</var>'s scroll inline offset.
+		1. For each focusgroup <var>group</var> containing one or more [=scroll marker control=] elements whose {{HTMLLinkElement/scrollTargetElement}} is a descendant of [=scroll container=]:
+
+			1. Let <var>markers</var> be all of the [=scroll marker control=] elements which are a part of the [=scroll marker group=] for the [=scroll container=].
+			1. Let <var>targets</var> be the set of {{HTMLLinkElement/scrollTargetElement}} for those controls sorted in [=tree order=].
+			1. Let <var>selected</var> be the first element in <var>targets</var>, or null if <var>targets</var> is empty.
+			1. For each <var>target</var> in <var>targets</var>:
+				1. Set <var>selected</var> to <var>target</var>.
+				1. Let <var>targetPosition</var> be the position that would be scrolled to if we scroll <var>target</var> into view.
+				1.	: If <var>targetPosition</var>'s scroll block offset is less than or equal to <var>position</var>'s scroll block offset, and
+						<var>targetPosition</var>'s scroll inline offset is less than or equal to <var>position</var>'s scroll inline offset.
+					::
+						Update <var>selected</var> to <var>target</var>.
+						Break.
+			1. 	: If <var>selected</var> is not null,
 				::
-					Update <var>selected</var> to <var>target</var>.
-					Break.
-		1. 	: If <var>selected</var> is not null,
-			::
-				1. Let <var>marker</var> be the first control in <var>markers</var> whose {{HTMLButtonElement/scrollTargetElement}} is <var>selected</var>.
-				1. Set the 'checked' state of <var>marker</var> to true.
-			: Otherwise,
-			::
-				Set the 'checked' state of all controls in <var>markers</var> to false.
+					1. Let <var>marker</var> be the first control in <var>markers</var> whose {{HTMLLinkElement/scrollTargetElement}} is <var>selected</var>.
+					1. Set the 'checked' state of <var>marker</var> to true.
+					1.	: If the active element was the <a href="https://open-ui.org/components/focusgroup.explainer/#last-focused-memory">last-focused element</a> of the <var>group</var>,
+						::
+							Focus <var>marker</var>
+					1. Set the <a href="https://open-ui.org/components/focusgroup.explainer/#last-focused-memory">last-focused element</a> of the <var>group</var>
+						to <var>marker</var>.
+				: Otherwise,
+				::
+					Set the 'checked' state of all controls in <var>markers</var> to false.
 	</div>
 
 	Issue: Should we allow for none of the markers to be currently active, e.g. if not yet scrolled past the position of the first marker.
```

Comment:
This already happens in this algorithm, right? I think you can remove this issue now.

Reply:
So in the previous algorithm we always select at least one marker. In your suggested update we don't. I think using your suggestion for the updated algorithm is fine.

Reply:
I don't think that's true? You have <[`8d81d76` (#10617)](https://github.com/w3c/csswg-drafts/pull/10617/commits/8d81d767074d6fb4c96624f8b5dbe33518dbb128#diff-fdbff78a41c06b32b51fc29f900df82da5efa11aa9e2e3394047d39c889569f9L961)> which explicitly handles the "selected is non-null" and has an "else" condition for when selected *is* null. The algorithm itself appears to have been able to fail to find a selection, too, if nothing in the list is before the target.

Oh, hm, I think I see the issue; the algo contradicts itself. It *always* sets `selected` to `target` at the start of the loop over `targets` (<[`8d81d76` (#10617)](https://github.com/w3c/csswg-drafts/pull/10617/commits/8d81d767074d6fb4c96624f8b5dbe33518dbb128#diff-fdbff78a41c06b32b51fc29f900df82da5efa11aa9e2e3394047d39c889569f9L954)>), but then it conditionally sets `selected` to `target` again later in the loop (<[`8d81d76` (#10617)](https://github.com/w3c/csswg-drafts/pull/10617/commits/8d81d767074d6fb4c96624f8b5dbe33518dbb128#diff-fdbff78a41c06b32b51fc29f900df82da5efa11aa9e2e3394047d39c889569f9L959)>) before breaking. Ah, and it also set `selected` to the first item in `targets` *before* the loop (<[`8d81d76` (#10617)](https://github.com/w3c/csswg-drafts/pull/10617/commits/8d81d767074d6fb4c96624f8b5dbe33518dbb128#diff-fdbff78a41c06b32b51fc29f900df82da5efa11aa9e2e3394047d39c889569f9L952)>)

So the "if non-null" check didn't do anything unless `targets` was *empty*; otherwise `selected` was guaranteed to be (redundantly) set to something in `targets`. And if I'm reading it correctly now, I think it would actually be guaranteed to only select either the first item in `targets` that is before the eventual position (if any are), or the last item (if nothing is), when it should have been selecting the *last* item that is before the eventual position.

Anyway, it's no problem if you *do* think one of them should be selected by default when they're all below the eventual position, but if so, I suspect the default should be the first rather than the last. But yeah, I'm inclined to go with "none are selected" in that case instead.